### PR TITLE
Refactor SHUFFLE_AGG satisfy SHUFFLE_JOIN

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Add SHUFFLE_AGG, SHUFFLE_JOIN satisfy ecah other.
And that can optimize plan:
```
      AGG（UPDATE FINALIZE)               AGG（UPDATE FINALIZE)  
              |                                   |             
            SHUFFLE           ====>               |             
              |                                   |             
            JOIN                                JOIN            
            /   \                               /   \           

```

```
          JOIN                               
        /      \                   JOIN      
       /        \                /      \    
    Shuffle   Shuffle   ==>     /        \   
      |          |            AGG        AGG 
      |          |                           
     AGG        AGG               
```

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 45 / 48 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/OutputPropertyDeriver.java|7|9|77.78%|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/HashDistributionDesc.java|6|7|85.71%|
|:large_blue_circle:|com/starrocks/sql/optimizer/RequiredPropertyDeriver.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/PropertyDeriverBase.java|20|20|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/HashDistributionSpec.java|2|2|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/HashDistributionDesc.java [hunk 1]
```diff
@@ -49,6 +53,15 @@ public boolean isSatisfy(HashDistributionDesc item) {
             return false;
         }
 
+        if (this.sourceType == SourceType.SHUFFLE_AGG && item.sourceType == SourceType.SHUFFLE_JOIN) {
+            return this.columns.size() == item.columns.size() && this.columns.equals(item.columns);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/HashDistributionDesc.java:57]: ```this.columns.equals(item.columns)``` is enough？

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 69 / 73 (94.52%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/OutputPropertyDeriver.java|7|9|77.78%|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/HashDistributionDesc.java|6|7|85.71%|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/HashDistributionSpec.java|26|27|96.30%|
|:large_blue_circle:|com/starrocks/sql/optimizer/RequiredPropertyDeriver.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/PropertyDeriverBase.java|20|20|100.00%|

**Comment by Seaven** [comment]
run starrocks_be_unittest

**Review by Youngwb — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**